apps/evm/src/clients/api/queries/useGetPools/getPools/index.ts

```diff
@@ -30,87 +28,101 @@ const safeGetIsolatedPoolParticipantCount = async (
 };
 
 export const getPools = async ({
+  publicClient,
   chainId,
   accountAddress,
-  primeContract,
-  poolLensContract,
-  legacyPoolComptrollerContract,
-  vaiControllerContract,
-  venusLensContract,
-  provider,
+  primeContractAddress,
+  poolLensContractAddress,
+  legacyPoolComptrollerContractAddress,
+  vaiControllerContractAddress,
+  venusLensContractAddress,
   tokens,
 }: GetPoolsInput) => {
   const [
     { pools: apiPools },
     isolatedPoolParticipantCounts,
-    { blockNumber: currentBlockNumber },
+    currentBlockNumber,
     unsafePrimeVTokenAddresses,
     primeMinimumXvsToStakeMantissa,
     userPrimeToken,
   ] = await Promise.all([
     getApiPools({ chainId }),
     safeGetIsolatedPoolParticipantCount({ chainId }),
     // Fetch current block number
-    getBlockNumber({ provider }),
+    publicClient.getBlockNumber(),
     // Prime related calls
-    primeContract?.getAllMarkets(), // TODO: get from API
-    primeContract?.MINIMUM_STAKED_XVS(), // TODO: get from API
-    accountAddress ? primeContract?.tokens(accountAddress) : undefined,
+    primeContractAddress
+      ? publicClient.readContract({
+          abi: primeAbi,
+          address: primeContractAddress,
+          functionName: 'getAllMarkets',
+        }) // TODO: get from API
+      : undefined,
+    primeContractAddress
+      ? publicClient.readContract({
+          abi: primeAbi,
+          address: primeContractAddress,
+          functionName: 'MINIMUM_STAKED_XVS',
+        }) // TODO: get from API
+      : undefined,
+    accountAddress && primeContractAddress
+      ? publicClient.readContract({
+          abi: primeAbi,
+          address: primeContractAddress,
+          functionName: 'tokens',
+          args: [accountAddress as Address],
+        })
+      : undefined,
   ]);
 
   const primeVTokenAddresses = unsafePrimeVTokenAddresses ?? [];
-  const isUserPrime = userPrimeToken?.exists || false;
+  const isUserPrime = userPrimeToken?.[0] || false;
 
   let userCollateralVTokenAddresses: string[] | undefined;
-  let userVTokenBalances:
-    | Awaited<ReturnType<PoolLens['callStatic']['vTokenBalancesAll']>>
-    | undefined;
   let userTokenBalances: TokenBalance[] | undefined;
+  let userVTokenBalances: VTokenBalance[] | undefined;
   let userVaiBorrowBalanceMantissa: BigNumber | undefined;
   let userPrimeApyMap: Map<string, PrimeApy> | undefined;
 
   if (accountAddress) {
     const [userCollaterals, userBalances, userVaiBorrowBalance, userPrimeApys] = await Promise.all([
       getUserCollateralAddresses({
         chainId,
-        accountAddress,
-        legacyPoolComptrollerContract,
+        accountAddress: accountAddress as Address,
+        legacyPoolComptrollerContractAddress,
         apiPools,
-        provider,
+        publicClient,
       }),
       getUserTokenBalances({
-        accountAddress,
+        accountAddress: accountAddress as Address,
```

Comment:
Wouldn't it be better to cast to `Address` in the hook and have the account address be of type `Address | undefined` here? Would avoid overriding the type in multiple lines

Reply:
Yeah makes sense. I originally kept the implementation as it is because it requires a lot of changes to the codebase, but I think we should make the change at some point anyway so why not do it in this PR.